### PR TITLE
test: To infinity and beyond! (adding `prop_{f32,f64}_{min_inf,max_inf,nan}` synthetics)

### PR DESCRIPTION
**Pull request by CommanderStorm**
image

https://www.youtube.com/watch?v=XDC74LEZkE8

### CODE REVIEW

File: java/mlt-tools/src/main/java/org/maplibre/mlt/tools/SyntheticMltGenerator.java [hunk 1]
```diff
@@ -139,11 +123,17 @@ private static void generateProperties() throws IOException {
...
     write("prop_f32_max", feat(p0, prop("val", Float.MAX_VALUE)), cfg());
+    write("prop_f32_pos_inf", feat(p0, prop("val", Float.POSITIVE_INFINITY)), cfg());
+    write("prop_f32_nan", feat(p0, prop("val", Float.NaN)), cfg());
     write("prop_f64", feat(p0, prop("val", (double) 3.141592653589793)), cfg());
+    write("prop_f64_neg_inf", feat(p0, prop("val", Double.NEGATIVE_INFINITY)), cfg());
     write("prop_f64_min", feat(p0, prop("val", Double.MIN_VALUE)), cfg());
-    // FIXME: fails in Java
-    // write("prop_f64_max", feat(p0, prop("val", Double.MAX_VALUE)), cfg());
+    write("prop_f64_max", feat(p0, prop("val", Double.MAX_VALUE)), cfg());
+    // FIXME: Produces the same output as prop_f64_max
```
Comment (CommanderStorm) [inline java/mlt-tools/src/main/java/org/maplibre/mlt/tools/SyntheticMltGenerator.java]:
```suggestion
    // FIXME: Fails in Java as it Produces the same output as prop_f64_max
```

**Comment by nyurik** [comment]
do we actually want to allow infinity / nans here?  I am not sure we support that in MVT, and for tag 0x01, we shouldn't add any new functionality if it is not already present in MVT

**Comment by CommanderStorm** [comment]
> do we actually want to allow infinity / nans here? I am not sure we support that in MVT, and for tag 0x01, we shouldn't add any new functionality if it is not already present in MVT

I don't know.
I did this because writing the tests for edge cases is on the bucket list for #836
These are defintive edge cases..

**Comment by nyurik** [comment]
heh, well, i guess it does belong because MVT uses https://github.com/mapbox/vector-tile-spec/blob/5330dfc6ba2d5f8c8278c2c4f56fff2c7dee1dbd/2.1/vector_tile.proto#L20 for float properties:

```protobuf
message Value {
        optional string string_value = 1;
        optional float float_value = 2;
        optional double double_value = 3;
```

and protobuf does define them to be valid. I have no idea how style spec might handle them (cc: @HarelM ?), but if they exist in MVT, they would need to be supported

File: rust/mlt-nom/tests/synthetics.rs [hunk 2]
```diff
@@ -18,10 +19,55 @@ fn test_one(mlt: &Path, json: &Path) {
...
+    );
+}
+
+/// Replace extremely small float values (< 1e-40) with 0.0 to handle codec precision issues
+fn normalize_tiny_floats(value: &Value) -> Value {
+    match value {
+        Value::Number(n) => {
+            if let Some(f) = n.as_f64() {
+                let eps = f64::from(f32::EPSILON);
+                if f.abs() < eps && f != 0.0 {
```
Comment (nyurik) [inline rust/mlt-nom/tests/synthetics.rs]: not sure - do you need to check if this is a real number, i.e. not a NaN or Inf?   Also, why do you need !=0.0 check?

File: rust/mlt-nom/src/geojson.rs [hunk 3]
```diff
@@ -48,7 +48,7 @@ impl FeatureCollection {
             for i in 0..geom.vector_types.len() {
                 let id = ids.and_then(|v| v.get(i).copied().flatten()).unwrap_or(0);
                 let geometry = geom.to_geojson(i)?;
-                let mut properties = HashMap::new();
+                let mut properties = BTreeMap::new();
```
Comment (nyurik) [inline rust/mlt-nom/src/geojson.rs:51]: why do you need btreemap here? we shouldn't be preserving order

Comment (CommanderStorm) [reply rust/mlt-nom/src/geojson.rs:51]: because otherwise we serialize to JSON in random order.
Not what we want..

We want our keys to be sorted actually

File: rust/mlt-nom/tests/synthetics.rs (hunk 2 above)
Comment (CommanderStorm) [reply rust/mlt-nom/tests/synthetics.rs]: yes, good catch
